### DISCUSSION TITLE
Explore a new way of testing CDI components in Quarkus

### DISCUSSION THREAD

**Discussion by giscus** [quarkusio/quarkus#34640] [Quarkus Blog/Website]
# Explore a new way of testing CDI components in Quarkus

Quarkus: Supersonic Subatomic Java

https://quarkus.io/blog/quarkus-component-test/

**Comment by bulldog98** [comment]
I think for most cases it's easier to just do the injections in the constructor and write a simple unit test, but I have to try that out. For the events and observing of these I see that it's useful.

**Comment by mhagnumdw** [comment]
Does `PanacheMock.mock(Holiday.class);` only work with `@QuarkusTest`? I would like to use it with `@QuarkusComponentTest`, because in my tests I would not need Quarkus to upload the entire context, including datasource etc. I need to mock `Holiday.persist`.
